src/passes/CoalesceLocals.cpp

```diff
@@ -440,7 +440,7 @@ void CoalesceLocals::applyIndices(std::vector<Index>& indices, Expression* root)
   auto numParams = getFunction()->getNumParams();
   Index newNumLocals = 0;
   for (auto index : indices) {
-    newNumLocals = std::max(newNumLocals, index + 1);
+    newNumLocals = std::max((Index::value_type)newNumLocals, index + 1);
```

Comment:
`index` is an `Index` but `index + 1` is a `uint32_t` because there's no `operator+` overload. So it can't pick the right template for `std::max` because the arguments don't match. So you can cast `newNumLocals` to `uint32_t` or cast `index + 1` to `Index` (in which case they will both implicitly convert to `uint32_t`).